Add Schema and code for storing daily view on user profiles

shyam0705 opened the pull request:
## Fixes issue #2582 

## Changes proposed
Added Schema and code for storing daily views on user profiles and added separate section for showing the most popular profiles of the last 28 days.
<!-- List all the proposed changes in your PR -->

<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->
<!--
[x] - Correct; marked as done
[X] - Correct; marked as done

[ ] - Not correct; marked as **not** done
-->

## Check List (Check all the applicable boxes) <!-- Follow the above conventions to check the box -->

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

<!-- Add all the screenshots which support your changes -->

## Note to reviewers

<!-- Add notes to reviewers if applicable -->


<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2691"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

eddiejaoude commented on models/ProfileStats.js:
```diff
@@ -0,0 +1,26 @@
+    required: true,
+  },
+  views: {
+    type: [viewSchema],
```
I think simplifying the model will make our code cleaner, similar to the way links are done https://github.com/EddieHubCommunity/LinkFree/blob/main/models/Link.js

shyam0705 replied:
do you suggesting schema like { date, profile views} ?

eddiejaoude replied:
Yes I think so. Here it is formatted for others to be able to read...

```js
new mongoose.Schema({
   username: String,
   views: {
      type: Number,
      default: 0,
   },
   date:{
      type: Date,
      default: new Date(new Date().toLocaleDateString()),
   }
});
```

It is probably worth creating a relationship to the profile with

```js
   profile: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "profile",
    },
```

shyam0705 replied:
Thanks for the suggestion, @eddiejaoude. Working on it

shyam0705 commented:
@eddiejaoude I have simplified the schema and added a section showing popular profiles in last 28 days, and the project is running fine locally.

UI:-

<img width="955" alt="image" src="https://user-images.githubusercontent.com/72148710/211142913-585bff4f-2492-4920-ba43-6ef046a0e687.png">

Database:-

<img width="757" alt="image" src="https://user-images.githubusercontent.com/72148710/211143034-2f771de3-bf2e-446c-858b-6175676bfe3a.png">

I don't know why one test is failing here when I tested locally, that test is passing correctly.

<img width="837" alt="image" src="https://user-images.githubusercontent.com/72148710/211143082-4adc0af6-7725-4fc9-80a5-72c3ea95c658.png">

eddiejaoude commented:
I think this is a flakey test, let me take a look

eddiejaoude reviewed (approved):
Looks good 👍 

I will merge into a temporary branch, and combine on the new page `Discover` that replaces the `Popular` profiles page
